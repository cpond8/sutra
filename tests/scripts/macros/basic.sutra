;; Basic macro definition and invocation tests
;; Tests minimal macro functionality and parameter passing

;; Test: Macro with parameter
(define (greet name) (print name))
(greet "Alice")
(greet "Bob")

;; Test: Macro with different parameter
(define (show value) (print value))
(show "test message")
(show 42)

;; Test: Definition-only macro (not invoked)
(define (unused_macro param) (print param))
