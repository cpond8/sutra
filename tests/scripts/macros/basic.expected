Alice
Bob
test message
42
